application/MainWindow.cpp

```diff
@@ -1563,95 +1491,40 @@ void MainWindow::on_actionLaunchInstanceOffline_triggered()
 void MainWindow::doLaunch(bool online, BaseProfilerFactory *profiler)
 {
 	if (!m_selectedInstance)
+	{
 		return;
+	}
 
-	// Find an account to use.
-	std::shared_ptr<MojangAccountList> accounts = MMC->accounts();
-	MojangAccountPtr account = accounts->activeAccount();
-	if (accounts->count() <= 0)
+	while (true)
```

Comment:
The `while(true)` loop is bad indeed, but something I had to do to shoehorn the MojangAuthSession thingy into the picture. Not being able to exit is bad too. I'll see if I can clean this up.